mycore-validation/src/main/java/org/mycore/validation/pdfa/MCRPDFAFunctions.java

```diff
@@ -125,11 +130,99 @@ private static Document createXML(String objectId, Map<String, ValidationResult>
      * @param resultEntry     A map entry containing the file name and its validation result.
      */
     private static void createXMLTag(Element derivateElement, Document document,
-        Map.Entry<String, ValidationResult> resultEntry) {
-        String fileName = resultEntry.getKey();
-        ValidationResult result = resultEntry.getValue();
-        Element fileElement = createFileElement(document, fileName, result);
-        derivateElement.appendChild(fileElement);
+        PDFAValidationResult resultEntry) {
+        String fileName = resultEntry.name();
+        ValidationResult result = resultEntry.result();
+        if (result != null) {
+            Element fileElement = createFileElement(document, fileName, result);
+            derivateElement.appendChild(fileElement);
+        } else {
+            Element fileElement = document.createElement("file");
+            fileElement.setAttribute("name", fileName);
+            derivateElement.appendChild(fileElement);
+            fileElement.setAttribute("flavour", "Validation Error");
+            List<Element> exceptions = createExceptionElements(document, resultEntry.exception());
+            Element exceptionElements = document.createElement("exceptions");
+            for (Element exception : exceptions) {
+                exceptionElements.appendChild(exception);
+            }
```

Comment:
You hand over an Exception and receive a List of Exception Elements? I think only one exception could be thrown finally and caught. So no need for "<exceptions>" or a List return type.